### PR TITLE
Set limits for how much memory docker build can use

### PR DESCRIPTION
**Pull request by yuvipanda**
This prevents builds from gobbling all the RAM on a node!

### CODE REVIEW

File: tests/memlimit/dockerfile/postBuild [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env python3
+"""
+Simplest program that tries to allocate a large amount of RAM.
+
+malloc lies on Linux by default, so we use memset to force the
+kernel to actually give us real memory.
```
Comment (betatim) [inline tests/memlimit/dockerfile/postBuild]: sneaky kernel!

File: repo2docker/app.py [hunk 2]
```diff
@@ -347,7 +364,7 @@ def start(self):
             if self.build:
                 self.log.info('Using %s builder\n', bp.name,
                               extra=dict(phase='building'))
-                for l in picked_buildpack.build(self.output_image_spec):
+                for l in picked_buildpack.build(self.output_image_spec, self.build_memory_limit):
```
Comment (betatim) [inline repo2docker/app.py]: should we make this a named argument? Just thinking we might want to add a CPU limit or other limits in the future and then I won't be able to remember in which order args go.

Comment (yuvipanda) [reply repo2docker/app.py]: Docker doesn't support CPU limits for builds yet, so IMO it's ok right now for this to be just a param that must be specified. We can always just screw ordering and say memory_limit=<whatever> when calling it regardless of it having a default value or not when we have too many of them I think?

**Comment by betatim** [comment]
Merged!

**Comment by betatim** [comment]
https://travis-ci.org/jupyter/repo2docker/jobs/309908787 this is the build for the merge. Deploy failed. :-/

**Pull request merged**